tools/parquetDDL/main.cpp

```
@@ -0,0 +1,184 @@
+      // or maybe here we need to load this column data to get the maximum length
+      // return "TINYINT";
+      return "VARCHAR(2000)";
+    }
+    case arrow::Type::type::BINARY:
+    {
+      return "VARCHAR(8000)";
+    }
+    case arrow::Type::type::FIXED_SIZE_BINARY:
+    {
+      std::shared_ptr<arrow::FixedSizeBinaryType> fType = std::static_pointer_cast<arrow::FixedSizeBinaryType>(dataType);
+      int byteWidth = fType->byte_width();
+      return "CHAR(" + std::to_string(byteWidth) + ")";
+    }
+    case arrow::Type::type::DATE32:
+    {
+      return "DATE";
+    }
+    case arrow::Type::type::DATE64:
+    {
+      return "DATE";
+    }
+    case arrow::Type::type::TIMESTAMP:
+    {
+      return "TIMESTAMP";
+    }
+    case arrow::Type::type::TIME32:
+    {
+      return "TIME";
+    }
+    case arrow::Type::type::DECIMAL128:
+    {
+      // get precision and scale
+      std::shared_ptr<arrow::DecimalType> fType = std::static_pointer_cast<arrow::DecimalType>(dataType);
+      int32_t fPrecision = fType->precision();
+      int32_t fScale = fType->scale();
+
+      return "DECIMAL(" + std::to_string(fPrecision) + "," + std::to_string(fScale) + ")";
+    }
+    default:
+      break;
+  }
+  return "FAULT";
+}
+
+void generateDDL(std::string filePth, std::string targetPth, std::string tableName)
+{
+  std::shared_ptr<arrow::Schema> parquetSchema;
+  getSchema(filePth, &parquetSchema);
+  std::vector<std::string> parquetCols;
+  std::vector<std::string> parquetTypes;
+
+  int fieldsNum = parquetSchema->num_fields();
+  for (int i = 0; i < fieldsNum; i++)
+  {
+    const std::shared_ptr<arrow::Field> tField = parquetSchema->field(i);
+    const std::string tName = tField->name();
+    auto tType = tField->type();
+    parquetCols.push_back(tName);
+    std::string colType = convert2mcs(tType, tType->id());
+    if (colType == "FAULT")
+    {
+      // emit fault error
+      std::cout << "Not allowed data type" << std::endl;
+      return;
+    }
```

`convert2mcs()` returning "FAULT" for a not allowed data type is not a robust approach. The program should return an error code and display an appropriate error message to the client, such as "Not allowed data type: DECIMAL256" or "Not allowed data type: List" etc. The not allowed type should be set in `convert2mcs()`.